src/api/utils/request_api.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Request is a a custom Network Layer.  Its main benefits of using it over fetch include
+ * a longer chain of promises which allows us to write error handling code all in one place.
+ * Request will give json back or some standard, meaningful error so that the errors and parsing
+ * don't have to be handled higher up in the call stack.
+ */
+
+class Request {
+  static headers() {
+    return {
+      'Content-Type': 'application/json'
+    }
+  }
+
+  static get(url) {
+    return this.fetchWrapper(url, null, 'GET')
+  }
+  static post(url, params) {
+    return this.fetchWrapper(url, params, 'POST')
+  }
+
+  static isJSON(response) {
+    // returns true if response is of type JSON
+    const contentType = response.headers.get('content-type');
+
+    return contentType && contentType.includes('application/json');
+  }
+
+  /**
+   * @param {Response} response
+   * @return {Promise} 
+   */
+  static parseResponseBody(response) {
+    if (this.isJSON(response)) {
+      return response.json();
+    }
+
+    return response.text();
+  }
+
+  /**
+   * @param {String} url
+   * @param {Object?} params
+   * @param {String} verb
+   * @return {Promise} 
+   */
```

Comment:
Yeah, if we're gonna do stuff like this i think we should actually use flow. "Object" and "String" types don't really tell you anything and could prob be inferred by the parameter names anyway